### ISSUE TITLE
Application names including regex/wildcard characters

### ISSUE THREAD

**Issue by begna112** [mkht/DSCR_Application#8]
https://github.com/mkht/DSCR_Application/blob/18e2d6383e4c936292cd686c30b1c98be56f207e/DSCResources/cApplication/cApplication.psm1#L679-L682

I have an application which is named like `MyApp [domain.com-5EF64F32]`. This is an application from a relatively popular vendor and I have no influence over the naming. I believe it updates the CRC32 every update.

```
Get-ItemProperty : Cannot retrieve the dynamic parameters for the cmdlet. The specified wildcard character pattern is not valid: MyApp [domain.com-5EF64F32]
At line:4 char:83
+ ... stallRegMachine | ForEach-Object { Get-ItemProperty $_.PSPath } | Whe ...
+                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Get-ItemProperty], ParameterBindingException
    + FullyQualifiedErrorId : GetDynamicParametersException,Microsoft.PowerShell.Commands.GetItemPropertyCommand
```

Changing the above instances to a literalpath resolves this issue:
```Powershell
Get-ItemProperty -LiteralPath $_.PSPath
```

**Comment by mkht** [comment]
Hi @begna112 
Thank you for reporting  an issue.
I will investigate and fix this as soon as possible if it is necessary.

For clearing a point of the issue, please tell me what the parameters you provided to the resource.
I tried two configs below, but can't reproduce this issue.

### Test A (Not Use Regex)
```PowerShell
cApplication MyApp
{
    Name = 'MyApp [domain.com-5EF64F32]'
    Fuzzy = $false    # False is default value
    InstallerPath = 'C:\installer.exe'
    Arguments = '/silent'
}
```

### Test B (Use Regex)
```PowerShell
cApplication MyApp
{
    Name = 'MyApp \[domain\.com-.+\]'    #Needs escaping .[] , if we use regex
    Fuzzy = $true
    InstallerPath = 'C:\installer.exe'
    Arguments = '/silent'
}
```

**Comment by begna112** [comment]
```Powershell
    cApplication 2012x64Minimum
    {
        Ensure = "Present"
        Name = 'Microsoft Visual C++ 2012 x64 Minimum Runtime - 11.0.61030'
        Version = '11.0.61030'
        InstallerPath = 'C:\Temp\vcredist\3 vcredist_x64 visual c++ redistributable 2012.exe'
        Arguments = '/install /quiet /norestart'
    }
```

The problem is actually that I have *another* application installed which is named `MyApp [domain.com-5EF64F32]`. So cApplication fails in the Test script when it checks if the application is already installed.

**Comment by mkht** [comment]
Okay, I can reproduce this.

### Script for reproduce
```PowerShell
# This script is only for testing. Do not run it on your machine.
$RegPath = "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall"
$AppName = '[a.b-0]'
New-Item -Path (Join-Path $UninstallRegMachine $AppName)
New-ItemProperty -LiteralPath (Join-Path $UninstallRegMachine $AppName) -Name 'DisplayName' -Value 'MyApp' -PropertyType String

Configuration Test
{
    Import-DscResource -ModuleName DSCR_Application
    cApplication 2012x64Minimum {
        Ensure        = "Present"
        Name          = 'Microsoft Visual C++ 2012 x64 Minimum Runtime - 11.0.61030'
        Version       = '11.0.61030'
        InstallerPath = 'C:\Temp\vcredist\3 vcredist_x64 visual c++ redistributable 2012.exe'
        Arguments     = '/install /quiet /norestart'
    }
}

Test -OutputPath C:\dsc
Test-DscConfiguration -Path C:\dsc -Verbose
```

**Comment by mkht** [comment]
This issue has been fixed in [v3.1.1](https://www.powershellgallery.com/packages/DSCR_Application/3.1.1)
Please try it out. :-)

**Comment by begna112** [comment]
Looks like it's working. Thanks!
